components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/SubscriptionsApiServiceImpl.java

```diff
@@ -91,8 +91,10 @@ public Response blockSubscription(String subscriptionId, String blockState, Stri
 
                 String apiContext = "";
                 String apiVersion = "";
+
                 if (apiId instanceof APIIdentifier) {
-                    apiContext = apiProvider.getAPIContext((APIIdentifier) apiId);
+                    String uuid = ApiMgtDAO.getInstance().getUUIDFromIdentifier((APIIdentifier) apiId);
```

Comment:
This will return an invalid UUID if we have same identifier for two organization